drivers/homeless_summary_report/app/models/homeless_summary_report/report.rb

```diff
@@ -479,12 +455,20 @@ def title_for(household_category, demographic_category)
       # Work through all the SPM report variants, building up the `report_clients` as we go.
       run_spm.each do |household_category, spec|
         report = spec[:base_variant]
-        detail_variant_name = "spm_#{household_category}__all"
+
+        # Create lists of client IDs in each demographic variant
+        demographic_variant_clients = {}
+        spec[:variants].each do |demographic_category, sub_spec|
+          detail_variant_name = "spm_#{household_category}__#{demographic_category}"
+          demographic_variant_clients[detail_variant_name] = client_ids_for_demographic_category(spec, sub_spec)
+        end
```

Comment:
I think I like this.  When we'd embedded it below, it was probably re-calculating for every client. Now, calculate the lists, then look in them later.

Reply:
Yep that's why I moved it, it was super slow